javascriptv3/example_code/s3/actions/delete-objects.js

```diff
@@ -1,34 +1,78 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-import { fileURLToPath } from "url";
-
 // snippet-start:[s3.JavaScript.buckets.deletemultipleobjectsV3]
-import { DeleteObjectsCommand, S3Client } from "@aws-sdk/client-s3";
-
-const client = new S3Client({});
+import {
+  DeleteObjectsCommand,
+  S3Client,
+  S3ServiceException,
+  waitUntilObjectNotExists,
+} from "@aws-sdk/client-s3";
 
-export const main = async () => {
-  const command = new DeleteObjectsCommand({
-    Bucket: "test-bucket",
-    Delete: {
-      Objects: [{ Key: "object1.txt" }, { Key: "object2.txt" }],
-    },
-  });
+/**
+ * Delete multiple objects from an S3 bucket.
+ * @param {{ bucketName: string, keys: string[] }}
+ */
+export const main = async ({ bucketName, keys }) => {
+  const client = new S3Client({});
 
   try {
-    const { Deleted } = await client.send(command);
+    const { Deleted } = await client.send(
+      new DeleteObjectsCommand({
+        Bucket: bucketName,
+        Delete: {
+          Objects: keys.map((k) => ({ Key: k })),
+        },
+      }),
+    );
+    for (const key in keys) {
+      await waitUntilObjectNotExists(
+        { client },
+        { Bucket: bucketName, Key: key },
+      );
+    }
     console.log(
       `Successfully deleted ${Deleted.length} objects from S3 bucket. Deleted objects:`,
     );
     console.log(Deleted.map((d) => ` • ${d.Key}`).join("\n"));
-  } catch (err) {
-    console.error(err);
+  } catch (caught) {
+    if (
+      caught instanceof S3ServiceException &&
+      caught.name === "NoSuchBucket"
```

Comment:
Is there a `NoSuchKey` error code or equivalent to use here in case the provided keys do not exist?

Reply:
No. If you try to delete a key that's not there, the SDK treats it as though it's already deleted.

```
~/.../actions (corepyle/s3-audit *$) node delete-object.js --bucketName coreys-default-bucket --key my-key.txt
The object "my-key.txt" from bucket "coreys-default-bucket" was deleted, or it didn't exist.
```